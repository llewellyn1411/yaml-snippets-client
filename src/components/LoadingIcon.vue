<template>
    <div v-if="visible" class="loading loading-lg"></div>
</template>

<script>
export default {
    name: 'LoadingIcon',
    props: {
        visible: {
            type: Boolean,
            default: false
        }
    }
};
</script>
