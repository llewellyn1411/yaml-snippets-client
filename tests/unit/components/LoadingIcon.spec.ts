import { shallowMount } from '@vue/test-utils';
import LoadingIcon from '../../../src/components/LoadingIcon.vue';

describe( 'LoadingIcon.vue', () => {
    test( 'should render correctly', () => {
        const wrapper = shallowMount( LoadingIcon, { propsData: { visible: true } } );
        expect( wrapper.html() ).toMatchSnapshot();
    } );

    test( 'should render when visible prop is true', () => {
        const wrapper = shallowMount( LoadingIcon, { propsData: { visible: true } } );
        expect( wrapper.find( '.loading' ).exists() ).toBe( true );
    } );

    test( 'should not render when visible prop is false', () => {
        const wrapper = shallowMount( LoadingIcon, { propsData: { visible: false } } );
        expect( wrapper.find( '.loading' ).exists() ).toBe( false );
    } );
} );
