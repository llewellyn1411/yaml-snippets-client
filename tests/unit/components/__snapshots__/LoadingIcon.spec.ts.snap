// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIcon.vue should render correctly 1`] = `<div class="loading loading-lg"></div>`;
